src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  public async *fetchRecentCasts({ pageSize = 100 } = {}): AsyncGenerator<
+    v1Cast,
+    void,
+    undefined
+  > {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of casts (with refreshed auth if necessary)
+      const response = await this.apis.v1Cast.recentCasts(
+        viewer,
+        cursor,
+        pageSize
+      );
+
+      // yield current page of casts
+      yield* response.data.result.casts;
+
+      // prep for next page
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * A list of users in reverse chronological order based on sign up.
+   */
+  public async *fetchRecentUsers({ pageSize = 100 } = {}): AsyncGenerator<
+    User,
+    void,
+    undefined
+  > {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of casts (with refreshed auth if necessary)
+      const response = await this.apis.user.recentUsers(
+        viewer,
+        cursor,
+        pageSize
+      );
+
+      // Extract the users array from the response (or provide an empty array as a default)
+      const users =
+        response.data.result?.users != null ? response.data.result.users : [];
+
+      // Create an async iterable for the users
+      for (const user of users) {
+        yield user;
+      }
+
+      // prep for next page
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Fetch all likes by a given user.
+   */
+  public async *fetchUserCastLikes(
+    user: { fid: number },
+    { pageSize = 100 } = {}
+  ): AsyncGenerator<ReactionWithCastMeta, void, undefined> {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of likes
+      const response = await this.apis.user.userCastLikes(
+        user.fid,
+        viewer,
+        pageSize,
+        cursor
+      );
+
+      // Extract the likes array from the response (or provide an empty array as a default)
+      const likes =
+        response.data.result?.likes != null ? response.data.result.likes : [];
+
+      // Create an async iterable for the likes
+      for (const like of likes) {
+        yield like;
+      }
+
+      // prep for next page
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Gets the specified user via their FID (if found)
+   */
+  public async lookupUserByFid(fid: number): Promise<User | undefined> {
```

Comment:
This is my fault, but while we're making changes we should probably be consistent across the different methods as to whether a function takes `fid: number` or `user: {fid: number}` as a parameter

Reply:
👍